src/main/java/com/zaxxer/hikari/HikariConfig.java

```diff
@@ -66,6 +67,7 @@ public class HikariConfig implements HikariConfigMXBean
    private volatile long idleTimeout;
    private volatile long leakDetectionThreshold;
    private volatile long maxLifetime;
+   private volatile double maxLifetimeVariance;
```

Comment:
I personally **really** try to avoid floating point numbers as much as I can, as they cause a lot of trouble. ( [What Every Computer Scientist Should Know About Floating-Point Arithmetic )
](https://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html)

I understand why you went for them: you need to be able to represent a value like `2.5%`, and it has a decimal point, which makes them not representable by integers.

I have a suggestion. How about you use [per mille](https://en.wikipedia.org/wiki/Per_mille) to represent the variance instead of percent. That way, you could change your type to `int`, which is a lot more friendly than `double`, and you'll be able to represent `2.5%` as `25‰`. If you think that we'll need more precision than that, you can change the type to make it more precise. Like "per million" for example. Though IMO "per mille" is enough.

Please let me know your thoughts.

Reply:
Per mille will do fine, agree